govcd/user.go

```diff
@@ -0,0 +1,463 @@
+
+	return user.Update()
+}
+
+// ChangePassword changes user's password
+// Constraints: the password must be non-empty, with a minimum of 6 characters
+func (user *OrgUser) ChangePassword(newPass string) error {
+	util.Logger.Printf("[TRACE] Changing user's password user: %s", user.User.Name)
+
+	user.User.Password = newPass
+
+	return user.Update()
+}
+
+// Enable enables a user if it was disabled. Fails otherwise.
+func (user *OrgUser) Enable() error {
+	util.Logger.Printf("[TRACE] Enabling user: %s", user.User.Name)
+
+	if user.User.IsEnabled {
+		return fmt.Errorf("user %s is already enabled", user.User.Name)
+	}
+	user.User.IsEnabled = true
+
+	return user.Update()
+}
+
+// Unlock unlocks a user that was locked out by the system.
+// Note that there is no procedure to LOCK a user: it is locked by the system when it exceeds the number of
+// unauthorized access attempts
+func (user *OrgUser) Unlock() error {
+	util.Logger.Printf("[TRACE] Unlocking user: %s", user.User.Name)
+
+	if !user.User.IsLocked {
+		return fmt.Errorf("user %s is not locked", user.User.Name)
+	}
+	user.User.IsLocked = false
+
+	return user.Update()
+}
+
+// ChangeRole changes a user's role
+// Fails is we try to set the same role as the current one.
+// Also fails if the provided role name is not found.
+func (user *OrgUser) ChangeRole(roleName string) error {
+	util.Logger.Printf("[TRACE] Changing user's role: %s", user.User.Name)
+
+	if roleName == "" {
+		return fmt.Errorf("role name cannot be empty")
+	}
+
+	if user.User.Role != nil && user.User.Role.Name == roleName {
+		return fmt.Errorf("new role is the same as current role")
+	}
+
+	newRole, err := user.adminOrg.GetRole(roleName)
+	if err != nil {
+		return err
+	}
+	user.User.Role = newRole
+
+	return user.Update()
+}
+
+// TakeOwnership takes ownership of the user's objects.
+// Ownership is transferred to the caller.
+// This is a call to make before deleting. Calling user.SafeDelete() will
+// run TakeOwnership before the actual user removal.
+// API Documentation: https://code.vmware.com/apis/442/vcloud-director#/doc/doc/operations/POST-TakeOwnership.html
+func (user *OrgUser) TakeOwnership() error {
+	util.Logger.Printf("[TRACE] Taking ownership from user: %s", user.User.Name)
+
+	userHREF, err := url.ParseRequestURI(user.User.Href + "/action/takeOwnership")
+	if err != nil {
+		return fmt.Errorf("error getting HREF for user %s : %v", user.User.Name, err)
+	}
+	util.Logger.Printf("[TRACE] Url for taking ownership from user : %#v and name: %s", userHREF, user.User.Name)
+
+	return user.client.ExecuteRequestWithoutResponse(userHREF.String(), http.MethodPost,
+		types.MimeAdminUser, "error taking ownership from user : %s", nil)
+}
+
+// validateUserForInput makes sure that the minimum data
+// needed for creating an org user has been included in the configuration
+func validateUserForCreation(user *types.User) error {
+	var missingField = "missing field %s"
+	if user.Xmlns == "" {
```

Comment:
I'm thinking if `CreateUser` could implicitly set this field. It's not going to change and not user dependent anyway.
Probably it would be the same for `type` or are there any chances these could differ?

Reply:
it actually could.